### DISCUSSION TITLE
a correct way to load an anonymous component?

### DISCUSSION THREAD

**Discussion by danielzzz** [spatie/laravel-package-tools#42] [Q&A]
hi,  I have troubles loading an anonymous component from a package in the main application
this is my config:

```
$package
            ->name('testpkg')
            ->hasViews();
```
the component file is at
```
resources/views/components/testcomponent.blade.php
```

I have tried:
```
<x-testpkg::testcomponent />
<x-testpkg-testcomponent />
```

but it does not work..

thanks, cheers
dan

**Comment by titantwentyone** [comment]
Apolgies for jumping into this but did you find a solution? Seems the only way is to create a component class but it's really not needed in my use case.

**Comment by danielzzz** [comment]
not that I remember.. sorry!

**Reply by titantwentyone** [reply]
Appreciate the response. Had to abandon usage of this for my particular package. May look into a PR.

**Comment by ousid** [comment]
A bit late but, I think you need to register your component in the `bootingPackage` method, see the following example:

```php
//...
use Illuminate\Support\Facades\Blade;

class YourServiceProvider extends PakcageServiceProvider
{
  public function bootingPackage()
  {
     Blade::component('testpkg::components.testcomponent');
  }
}
```

For more info, take a look at [Jetstream](https://github.com/laravel/jetstream) service Provider

https://github.com/laravel/jetstream/blob/f5ae2a6ff2d48c026b415d02369cb463322968d7/src/JetstreamServiceProvider.php#L103

**Comment by legecha** [comment]
From the [package development documentation](https://laravel.com/docs/10.x/packages#views):

> If your package contains anonymous components, they must be placed within a components directory of your package's "views" directory (as specified by the [loadViewsFrom method](https://laravel.com/docs/10.x/packages#views)). Then, you may render them by prefixing the component name with the package's view namespace:
> 
> `<x-courier::alert />`

In your service provider, where you define your `configurePackage()` method, as long as you call `$package->hasviews()` then the necessary `loadViewsFrom()` call will be handled (inside `PackageServiceProvider`).

So let's say, in your package, you have:

`resources/views/components/label.blade.php`

Then inside your service provider you have:

```
$package
    ->name('jeff')
    ->hasViews();
```

Then you will be able use the following in your project:

```
<x-jeff::label />
```
